invenio_rdm_records/services/results.py

```diff
@@ -69,3 +72,33 @@ def get_value_service(self, value):
     def pick(self, identity, resolved_rec):
         """Pick fields defined in the entity resolver."""
         return resolved_rec
+
+
+class RDMRecordList(RecordList):
+    """Record list with custom fields."""
+
+    @property
+    def hits(self):
+        """Iterator over the hits."""
+        for hit in self._results:
+            # Load dump
+            record_dict = hit.to_dict()
+            if record_dict["is_published"]:
```

Comment:
minor: wouldn't it be better to rely on the [`is_draft` property](https://github.com/inveniosoftware/invenio-drafts-resources/blob/116f25479db770312ee9672104d89a587c497cc1/invenio_drafts_resources/records/api.py#L63)? Though I think this should be a `ConstantField` systemfield so that it makes it to the indexed document.

Reply:
I think that's what caused the issue in the first place. the "search" method can be used for drafts or records, however, the method lives inside the record's service. So using `record.is_draft` always returns `False`.

Reply:
(@alejandromumo not for taking any action now, just commenting so that others might be able to move this forward)

I meant something like:

```suggestion
            if record_dict["is_draft"]:
```

You're right that accessing the `record_cls.is_draft` would always return `False`. But we should make sure also that whatever is indexed in the drafts index should have `is_draft: true` (I hope it does, otherwise it would be very confusing 😅)

Reply:
on my tests there exists no `is_draft` in `record_dict`